packages/cfd/src/Containers/cfd-password-modal.tsx

```diff
@@ -825,7 +825,9 @@ const CFDPasswordModal = observer(({ form_error, platform }: TCFDPasswordModalPr
             (values as TCFDPasswordFormValues & { platform: string }).platform = platform;
             submitCFDPassword(values, actions);
         }
-        closeModal();
+        if (product === 'zero_spread') {
+            setProduct();
+        }
```

Comment:
Why are you removing closeModal? and why are you specifically hardcoding zero_spread in the code?